perfschema/virtual_tables.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package perfschema
+
+import (
+	"fmt"
+	log "github.com/Sirupsen/logrus"
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/context"
+	"github.com/pingcap/tidb/model"
+	"github.com/pingcap/tidb/sessionctx/variable"
+	"github.com/pingcap/tidb/table"
+	"github.com/pingcap/tidb/table/tables"
+	"github.com/pingcap/tidb/util/types"
+)
+
+// session/global status decided by scope
+type statusDataSource struct {
+	meta        *model.TableInfo
+	cols        []*table.Column
+	globalScope bool
+}
+
+// GetRows implement the interface of VirtualDataSource
+func (ds *statusDataSource) GetRows(ctx context.Context) (fullRows [][]types.Datum,
+	err error) {
+	sessionVars := ctx.GetSessionVars()
+	statusVars, err := variable.GetStatusVars(sessionVars)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	rows := [][]types.Datum{}
+	for status, v := range statusVars {
+		if ds.globalScope && v.Scope == variable.ScopeSession {
```

Comment:
What if data source is session scope and v.Scop == variable.ScopeGlobal ?
And NoneScope seems should not be added too, I'm not sure about it.

Reply:
session status will show all status, including global andl local, global status will show global status only.
this detail is from [mysql document](https://dev.mysql.com/doc/refman/5.7/en/performance-schema-status-variable-tables.html)